src/public/cards/uploader.ts

```diff
@@ -0,0 +1,109 @@
+import { startWaiting, stopWaiting } from "../utils";
+
+/**
+ * All logic related to the first card (uploading .csv file)
+ */
+export class CSVUploader {
+  private container!: HTMLElement;
+  private fileInput!: HTMLInputElement;
+  private fileName!: HTMLElement;
+  private fileError!: HTMLElement;
+
+  init() {
+    this.container = document.getElementsByClassName(
+      "homeContainer"
+    )[0] as HTMLElement;
+    this.fileInput = document.getElementById("importFile") as HTMLInputElement;
+    this.fileName = document.getElementById("fileName") as HTMLElement;
+    this.fileError = document.getElementById("fileError") as HTMLElement;
+    this.hideError();
+    this.cleanState();
+    this.showCard();
+  }
+
+  /**
+   * Delete session storage and file uploaded to begin to upload
+   */
+  private cleanState() {
+    if (this.fileInput && this.fileInput.files) {
+      this.fileInput.value = "";
+    }
+    this.fileName.innerText = " ";
+    sessionStorage.removeItem("sourceKey");
+    sessionStorage.removeItem("rows");
+    sessionStorage.removeItem("headers");
+    sessionStorage.removeItem("entityName");
+  }
+
+  /**
+   * Show file name when user added a file
+   */
+  showFile() {
+    const files = this.fileInput?.files;
+    this.hideError();
+    if (files && files.length) {
+      this.fileName.innerText = files[0].name;
+      sessionStorage.setItem("entityName", files[0].name.replace(".csv", ""));
+    }
+  }
+
+  /**
+   * Read and save to session storage the file data
+   */
+  readFile() {
+    const params = new URLSearchParams(window.location.search);
+    const sourceKey = params.get("sourceKey");
+    if (!sourceKey) {
+      this.fileError.innerHTML = "No source key defined in URL";
+      this.showError();
+      throw Error("No source key defined in URL");
+    }
+    sessionStorage.setItem("sourceKey", sourceKey);
+
+    const files = this.fileInput?.files;
+    if (!files || !files.length || !files[0].name.endsWith(".csv")) {
+      this.fileError.innerHTML = "Select a valid file";
+      this.showError();
+      throw Error("Select a valid file");
+    }
+    startWaiting();
+    let fr = new FileReader();
+    fr.onload = (event) => {
+      stopWaiting();
+      if (event && event.target && event.target.result) {
+        const result = event.target.result as string;
+        const rows = result.split(/\r?\n|\r/);
```

Comment:
can you add comment to explain how we split the string

Reply:
legacy code, but will explain